### PR TITLE
coqPackages.trakt: init at 1.0, coqPackages.smtcoq: init at itp22

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 162265` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>coqPackages.smtcoq</li>
    <li>coqPackages.trakt</li>
  </ul>
</details>

File: pkgs/development/coq-modules/smtcoq/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
+{ lib, stdenv, mkCoqDerivation, coq, trakt, cvc4, veriT, version ? null }:
+with lib;
+
+mkCoqDerivation {
+  pname = "smtcoq";
+  owner = "smtcoq";
+
+  release."itp22".rev    = "1d60d37558d85a4bfd794220ec48849982bdc979";
+  release."itp22".sha256 = "sha256-CdPfgDfeJy8Q6ZlQeVCSR/x8ZlJ2kSEF6F5UnAespnQ=";
```
Comment (papojari) [inline pkgs/development/coq-modules/smtcoq/default.nix:9]:
```suggestion
  release."itp22" = {
    rev    = "1d60d37558d85a4bfd794220ec48849982bdc979";
    sha256 = "sha256-CdPfgDfeJy8Q6ZlQeVCSR/x8ZlJ2kSEF6F5UnAespnQ=";
  };
```
If `"itp22"` changes it won't have to be updated twice.

File: pkgs/development/coq-modules/smtcoq/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  defaultVersion = with versions; switch [ coq.version mathcomp.version ] [
+    { cases = [ (isGe "8.13") ]; out = "itp22"; }
+  ] null;
+
+  propagatedBuildInputs = [ trakt cvc4 ] ++ lib.optionals (!stdenv.isDarwin) [ veriT ];
+  extraNativeBuildInputs = with coq.ocamlPackages; [ ocaml ocamlbuild ];
+  extraBuildInputs = with coq.ocamlPackages; [ findlib num zarith ];
+
+  meta = {
+    description = "Communication between Coq and SAT/SMT solvers ";
```
Comment (papojari) [inline pkgs/development/coq-modules/smtcoq/default.nix:21]:
```suggestion
    description = "Communication between Coq and SAT/SMT solvers";
```
Unnecessary space.

File: pkgs/development/libraries/cln/default.nix [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
 stdenv.mkDerivation rec {
   pname = "cln";
   version = "1.3.6";
 
-  src = fetchurl {
-    url = "${meta.homepage}${pname}-${version}.tar.bz2";
-    sha256 = "0jlq9l4hphk7qqlgqj9ihjp4m3rwjbhk6q4v00lsbgbri07574pl";
+  src = fetchgit {
+    url = "git://www.ginac.de/cln.git";
```
Comment (papojari) [inline pkgs/development/libraries/cln/default.nix:8]:
```suggestion
    url = "https://www.ginac.de/cln.git";
```
`git://` is considered unsafe.

**Review by auroraanna — CHANGES_REQUESTED** [review]

**Comment by papojari** [comment]
Okay it doesn't work now. But I know why. I shouldn't have recommend the `https://` protocol in this case because that repo is only available via `git://`. I didn't check that. Some I read that he git protocol is unsafe. Now I read up on `git://` some more and the lack of authentication means that by default anyone can write to the repo. Fortunately the owners seem to have disabled that (I can't push to it) so it shouldn't be a problem. Sorry for the mess!

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 162265` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>echidna</li>
    <li>maude</li>
    <li>tamarin-prover</li>
  </ul>
</details>
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>ginac</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>cln</li>
    <li>coqPackages.smtcoq</li>
    <li>coqPackages.trakt</li>
    <li>cvc4</li>
  </ul>
</details>

**Comment by CohenCyril** [comment]
```
nix-build -A cvc4
these derivations will be built:
  /nix/store/s12fa4gvhv4nkll56nmhjm2dbm5v03y8-cln-1.3.6.drv
  /nix/store/lg77x1k703m73xdizw40nblmgyny57ic-cvc4-1.8.drv
these paths will be fetched (82.92 MiB download, 199.37 MiB unpacked):
  /nix/store/7hl64m5n08z8hqclm2mvqj4ka4i32q47-gnome-vfs-2.24.4
  /nix/store/ahl8r4p0cnqwrqapp5546gd7idz4vrhg-openjdk-8u322-ga
  /nix/store/bdxvdpcn56qam3wv0pm1nb39747a7dli-openjdk-8u322-ga-jre
  /nix/store/ixsi2hwc2w5awxkhjxqk4fr5lfa725v2-antlr-3.4
copying path '/nix/store/7hl64m5n08z8hqclm2mvqj4ka4i32q47-gnome-vfs-2.24.4' from 'https://cache.nixos.org'...
building '/nix/store/s12fa4gvhv4nkll56nmhjm2dbm5v03y8-cln-1.3.6.drv'...
unpacking sources
unpacking source archive /nix/store/3jzdpn5lg48dkm6ci6zyg0s8vfkd5r3h-cln
source root is cln
patching sources
configuring
no configure script, doing nothing
building
no Makefile, doing nothing
installing
install flags: SHELL=/nix/store/30j23057fqnnc1p4jqmq73p0gxgn0frq-bash-5.1-p16/bin/bash install
make: *** No rule to make target 'install'.  Stop.
builder for '/nix/store/s12fa4gvhv4nkll56nmhjm2dbm5v03y8-cln-1.3.6.drv' failed with exit code 2
cannot build derivation '/nix/store/lg77x1k703m73xdizw40nblmgyny57ic-cvc4-1.8.drv': 1 dependencies couldn't be built
error: build of '/nix/store/lg77x1k703m73xdizw40nblmgyny57ic-cvc4-1.8.drv' failed
```

**Comment by siraben** [comment]
@CohenCyril what system does this build log correspond to?

**Comment by CohenCyril** [comment]
NixOS x86_64

**Comment by CohenCyril** [comment]
On the master branch of nixpkgs

**Comment by siraben** [comment]
For some reason my patch works on darwin but causes linux to fail. Build error on darwin with the previous definition of `cln`: http://ix.io/3VeA

**Comment by hqurve** [comment]
After a bit of testing, it appears as if `fetchGit` is not working correctly with this site. By inspecting the `src` by running `nix build .#cln.src`, the directory structure is much more similar to the [master branch](https://www.ginac.de/CLN/cln.git/?p=cln.git;a=tree;h=refs/heads/master;hb=refs/heads/master) than it is to the [1.3.6 release](https://www.ginac.de/CLN/cln.git/?p=cln.git;a=commit;h=d4ba1cc869be2c647c4ab48ac571b1fc9c2021a9). Notice that the CMakeList.txt file is not present on 1.3.6.

I also tried supplying the commit hash, but to no avail

**Comment by gebner** [comment]
I'm really surprised that ofborg didn't fail CI here even though cln failed to build: https://logs.nix.ci/?key=nixos/nixpkgs.162265&attempt_id=b7315ae6-ac87-448d-a6b4-4bfc712ab818
```
error: build of '/nix/store/dqninbrx9wp5rln6amikn5w76xxr1lkl-coq8.13-smtcoq-itp22.drv', '/nix/store/s12fa4gvhv4nkll56nmhjm2dbm5v03y8-cln-1.3.6.drv' failed
```

**Comment by CohenCyril** [comment]
I didn't manage to compile smtcoq for coq 8.14 and 8.15 and [is documented as such in the release `.opam` file](https://github.com/smtcoq/smtcoq/blob/1d60d37558d85a4bfd794220ec48849982bdc979/coq-smtcoq.opam#L15). Did you ever test it for Coq 8.14 and 8.15?

**Pull request merged**